air/src/air.rs

```diff
@@ -47,10 +47,14 @@ pub trait Air<AB: AirBuilder>: BaseAir<AB::F> {
 /// as well cases where the constraints are evaluated on an evaluation trace and combined using randomness.
 pub trait AirBuilder: Sized {
     /// Underlying field type.
-    type F: Field;
+    type F: Field + Into<Self::I>;
+
+    /// An intermediary between F and Expr, usually equal to F.
+    /// Alternative use case: I = F::Packing, Expr = <EF as ExtensionField<F>>::ExtensionPacking
+    type I;
```

Comment:
Hmm what do you think about adding an Algebra bound here? (We don't need it for bounds but it makes logical sense and might be useful in some rare cases)

```suggestion
    type I: Algebra<F>;
```